### ISSUE TITLE
Wrong report of TS2525

### ISSUE THREAD

**Issue by tibor19** [Microsoft/TypeScript#21634]
**TypeScript Version:**  2.8.0-dev.20180204

**Search Terms:** 
TS2525  "Initializer provides no value for this binding element and the binding element has no default value"
**Code**
```ts
const [, a, b] = [, , ];

// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
```

**Expected behavior:**
Should compile without errors
**Actual behavior:**
Reports error TS2525. Adding an extra comma, makes the message go away.
```ts
const [, a, b] = [, , ,]; 
```
**Playground Link:**  

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
Trailing commas in array literals don't create an empty element. `[ , , ]` has length 2, not 3.
